crypto/digestible/src/lib.rs

```diff
@@ -49,128 +216,258 @@ pub trait Digestible {
 
-impl<T: RawDigestible> Digestible for T {
+impl<T: DigestibleAsBytes> Digestible for T {
     #[inline]
-    fn digest<D: Digest>(&self, hasher: &mut D) {
-        hasher.update(self)
+    fn append_to_transcript<DT: DigestTranscript>(
+        &self,
+        context: &'static [u8],
+        transcript: &mut DT,
+    ) {
+        <Self as AsRef<[u8]>>::as_ref(self).append_to_transcript(context, transcript);
     }
 }
 
 // Built-in byte arrays
-impl RawDigestible for [u8; 1] {}
-impl RawDigestible for [u8; 2] {}
-impl RawDigestible for [u8; 3] {}
-impl RawDigestible for [u8; 4] {}
-impl RawDigestible for [u8; 5] {}
-impl RawDigestible for [u8; 6] {}
-impl RawDigestible for [u8; 7] {}
-impl RawDigestible for [u8; 8] {}
-impl RawDigestible for [u8; 9] {}
-impl RawDigestible for [u8; 10] {}
-impl RawDigestible for [u8; 11] {}
-impl RawDigestible for [u8; 12] {}
-impl RawDigestible for [u8; 13] {}
-impl RawDigestible for [u8; 14] {}
-impl RawDigestible for [u8; 15] {}
-impl RawDigestible for [u8; 16] {}
-impl RawDigestible for [u8; 17] {}
-impl RawDigestible for [u8; 18] {}
-impl RawDigestible for [u8; 19] {}
-impl RawDigestible for [u8; 20] {}
-impl RawDigestible for [u8; 21] {}
-impl RawDigestible for [u8; 22] {}
-impl RawDigestible for [u8; 23] {}
-impl RawDigestible for [u8; 24] {}
-impl RawDigestible for [u8; 25] {}
-impl RawDigestible for [u8; 26] {}
-impl RawDigestible for [u8; 27] {}
-impl RawDigestible for [u8; 28] {}
-impl RawDigestible for [u8; 29] {}
-impl RawDigestible for [u8; 30] {}
-impl RawDigestible for [u8; 31] {}
-impl RawDigestible for [u8; 32] {}
-
-impl<Length: generic_array::ArrayLength<u8>> RawDigestible
-    for generic_array::GenericArray<u8, Length>
-{
-}
+impl DigestibleAsBytes for [u8; 1] {}
+impl DigestibleAsBytes for [u8; 2] {}
+impl DigestibleAsBytes for [u8; 3] {}
+impl DigestibleAsBytes for [u8; 4] {}
+impl DigestibleAsBytes for [u8; 5] {}
+impl DigestibleAsBytes for [u8; 6] {}
+impl DigestibleAsBytes for [u8; 7] {}
+impl DigestibleAsBytes for [u8; 8] {}
+impl DigestibleAsBytes for [u8; 9] {}
+impl DigestibleAsBytes for [u8; 10] {}
+impl DigestibleAsBytes for [u8; 11] {}
+impl DigestibleAsBytes for [u8; 12] {}
+impl DigestibleAsBytes for [u8; 13] {}
+impl DigestibleAsBytes for [u8; 14] {}
+impl DigestibleAsBytes for [u8; 15] {}
+impl DigestibleAsBytes for [u8; 16] {}
+impl DigestibleAsBytes for [u8; 17] {}
+impl DigestibleAsBytes for [u8; 18] {}
+impl DigestibleAsBytes for [u8; 19] {}
+impl DigestibleAsBytes for [u8; 20] {}
+impl DigestibleAsBytes for [u8; 21] {}
+impl DigestibleAsBytes for [u8; 22] {}
+impl DigestibleAsBytes for [u8; 23] {}
+impl DigestibleAsBytes for [u8; 24] {}
+impl DigestibleAsBytes for [u8; 25] {}
+impl DigestibleAsBytes for [u8; 26] {}
+impl DigestibleAsBytes for [u8; 27] {}
+impl DigestibleAsBytes for [u8; 28] {}
+impl DigestibleAsBytes for [u8; 29] {}
+impl DigestibleAsBytes for [u8; 30] {}
+impl DigestibleAsBytes for [u8; 31] {}
+impl DigestibleAsBytes for [u8; 32] {}
```

Comment:
Not 100% this will work, buuut:

```suggestion
impl<const N: usize> DigestibleAsBytes for [u8; N] {}
```

Reply:
will investigate, thanks :)

Reply:
it gave me this:

```
error[E0658]: const generics are unstable
   --> crypto/digestible/src/lib.rs:372:12
    |
372 | impl<const N: usize> DigestibleAsBytes for [u8; N] {}
    |            ^
    |
    = note: see issue #44580 <https://github.com/rust-lang/rust/issues/44580> for more information
    = help: add `#![feature(const_generics)]` to the crate attributes to enable

error[E0277]: arrays only have std trait implementations for lengths 0..=32
   --> crypto/digestible/src/lib.rs:372:22
    |
358 | pub trait DigestibleAsBytes: AsRef<[u8]> + Sized {}
    |                              ----------- required by this bound in `DigestibleAsBytes`
...
372 | impl<const N: usize> DigestibleAsBytes for [u8; N] {}
    |                      ^^^^^^^^^^^^^^^^^ the trait `core::array::LengthAtMost32` is not implemented for `[u8; N]`
    |
    = note: required because of the requirements on the impl of `core::convert::AsRef<[u8]>` for `[u8; N]`

error: aborting due to 2 previous errors
```

Reply:
i could add it, but right now, nothing is tying this crate to the nightly compiler, so i'd rather wait until that feature is stable